### PR TITLE
bumped dataframes version

### PR DESCRIPTION
**Pull request by ExpandingMan**
Don't think any of the changes should affect this package.

Note that, in the long run, it would probably be best to remove DataFrames as a dependency in favor of the Tables interface, but I didn't look into how involved that would be.

### CODE REVIEW

**Comment by cstjean** [comment]
I don't understand why https://travis-ci.org/github/cstjean/ScikitLearn.jl/jobs/744241946#L188 says v.0.22.0, but https://travis-ci.org/github/cstjean/ScikitLearn.jl/jobs/744241946#L296 says 0.21.8. And I don't get why it fails on 1.4 but succeeds on nightly 😢

**Comment by ExpandingMan** [comment]
Gah, this stuff is so annoying.  Presumably will need to at least wait for [this](https://github.com/JuliaStats/RDatasets.jl/pull/110).  Particularly aggravating because it's only used in tests.

**Comment by nalimilan** [comment]
RDatasets 0.7.2 now supports DataFrames 0.22.

**Comment by cstjean** [comment]
Ok, so there's some `df[:some_col]` [somewhere in the code](https://www.cssdm.gouv.qc.ca/wp-content/uploads/P2015-1-admission-transport.pdf#page=12) causing the failures, that should be fixed. What mystifies me is that the nightly tests pass, but that's probably because the resolver installed some old version of DataFrames.jl.

**Comment by ExpandingMan** [comment]
Ok, I found the failure and fixed it.  I've also taken the liberty to switch the latest travis run from 1.4 to 1.5.  Let's see if it passes.

File: .travis.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ env:
     - PYTHON=Conda
 julia:
     - 1.0
-    - 1.4
+    - 1.5
```
Comment (nalimilan) [inline .travis.yml:11]: Why not just use `1`?

Comment (ExpandingMan) [reply .travis.yml:11]: yeah, I keep forgetting that, I only recently realized that putting `1` gives latest `1`.  Let's see what the test result is and I'll put it in.

**Comment by ExpandingMan** [comment]
There was a failure on mac.  Looks full-blown dependency hell involving GaussianMixtures and StatsBase

**Comment by cstjean** [comment]
Offf, I don't know what the right solution is to this. GaussianMixtures is just a test dependency, so technically, 1.0 might still work perfectly well. Unless someone has a bright idea, I'd say we drop Julia 1.0 support in the name of sanity. 1.6 will be LTS anyway, right?

**Comment by nalimilan** [comment]
Looks like the resolver got lost in the requirements on Compat imposed by GaussianMixtures and ScikitLearn. Maybe you could try requiring `Compat = "3"` to make things simpler for it. Or even `Compat = "3.6"` like GaussianMixtures. Likewise, requiring `StatsBase = "0.33"` might help. Of course that's not ideal...

**Comment by ExpandingMan** [comment]
Alright, I tried that, let's see how it goes.

**Comment by nalimilan** [comment]
Failures are due to GaussianMixtures not supporting StatsBase 0.33. It supports it on master but no release has been tagged since then. I've asked at https://github.com/davidavdav/GaussianMixtures.jl/pull/68.

**Comment by nalimilan** [comment]
Though you could try tweaking only the Compat bounds instead.

**Comment by ExpandingMan** [comment]
Ok, I tried triggering this again (by bumping the PyCall version) and am just sort of hoping that everything miraculously works now because of changes in other packages.

**Comment by ExpandingMan** [comment]
@cstjean, I don't see much in the way of DataFrames here other than in `src/dataframes.jl`, is that correct or am I missing some?  I'm considering how much work it would be to drop DataFrames as a dependency altogether, and switch to the Tables.jl interface.  Would you be open to this?  (I don't *think* it would be breaking, but I'm not sure yet)

**Comment by cstjean** [comment]
Yes, the DataFrames support should be confined to `src/dataframes.jl`. If you're willing to do the work to get `Tables.jl` support working, and you believe that it's not (significantly?) breaking for users, then that sounds good to me.

**Comment by ExpandingMan** [comment]
I'm not yet confident that it won't be breaking, but I am confident it won't be very breaking.  Ideally would like to merge this PR before taking that on, but it's still breaking because GaussianMixtures needs a tag.

**Comment by cstjean** [comment]
> Ideally would like to merge this PR before taking that on, but it's still breaking because GaussianMixtures needs a tag.

If that would help, I would be fine merging this with `@test_broken` or somesuch temporary work-around.

**Comment by ExpandingMan** [comment]
It's up to you of course, but there's no good reason why the compat issue can't just be fixed, so maybe let's wait a bit longer.  I tried to make noise about this in various places today.

**Comment by nalimilan** [comment]
https://github.com/JuliaRegistries/General/pull/25016 was opened, but it requires manual merging. Make some noise there, that could help!

**Comment by nalimilan** [comment]
See also https://github.com/davidavdav/GaussianMixtures.jl/pull/81.

**Comment by nalimilan** [comment]
OK, GaussianMixtures has been tagged. Should be good to start another CI run.

**Comment by ExpandingMan** [comment]
Looks like it's passing.  For some reason travis now seems to wait for mac to finish before it runs linux, which I find rather irritating.

**Comment by ExpandingMan** [comment]
Ok, passing, finally.  @cstjean , please merge when you get a chance?

Finally I think this is the last major package holding back DataFrames 0.22

**Comment by cstjean** [comment]
Thank you for this PR!

**Comment by ExpandingMan** [comment]
No problem!  Also, don't forget we'll need a tag.  Thanks!

**Pull request merged**